Source/Core/Core/CoreTiming.cpp

```diff
@@ -392,9 +405,10 @@ void Advance()
 {
 	MoveEvents();
 
-	int cyclesExecuted = slicelength - PowerPC::ppcState.downcount;
+	int cyclesExecuted = slicelength - DowncountToCycles(PowerPC::ppcState.downcount);
 	globalTimer += cyclesExecuted;
-	PowerPC::ppcState.downcount = slicelength;
+	lastOCFactor = SConfig::GetInstance().m_OCEnable ? SConfig::GetInstance().m_OCFactor : 1.0f;
+	PowerPC::ppcState.downcount = CyclesToDowncount(slicelength);
```

Comment:
It would probably be a bit clearer to call the setting "IPCAdjustment", since it doesn't really have anything to do with the clock speed; it's basically just adjusting how many instructions the CPU executes per cycle.